kyaml/yaml/fns.go

```diff
@@ -221,6 +226,42 @@ func (e ElementMatcher) Filter(rn *RNode) (*RNode, error) {
 	return nil, nil
 }
 
+func GetElement(field string) ElementGetter {
+	return ElementGetter{FieldName: field}
+}
+
+// ElementGetter returns the first element from a Sequence which has
+// specified field name (key). Items in the list should be mapping
+type ElementGetter struct {
```

Comment:
As named, this should be the getter counterpart to ElementSetter, but it's not, and I find that confusing.

It seems ElementMatcher already plays the role of a counterpart to ElementSetter.

Can that function be used here?

If not, then lets try to think of a better name for this (something that doesn't make it look like the counterpart to ElementSetter).

Regardless, if you want to keep it, let's put it and the test for it in a separate PR and merge that first.

Reply:
ElementMatcher can only match a key-value pair and when it calls FieldMatcher it will always create a new value node for FieldMatcher. So it cannot be used to *get* an item with a specific key regardless the value from a slice. 

So I think the ElementGetter is actually the counterpart to ElementSetter.  Setter will set the value for the key. Getter will return the value pointed by key. What ElementMatcher does is returning the pair of key and value which match the given key and value.

Reply:
But I can try to add a new option in ElementMatcher to handle "no value specified" case. Will update the pr.